src/node/gov/handlers/service_state.h

```diff
@@ -0,0 +1,637 @@
+
+    return node;
+  }
+
+  void init_service_state_handlers(ccf::BaseEndpointRegistry& registry)
+  {
+    auto get_constitution = [&](auto& ctx, ApiVersion api_version) {
+      switch (api_version)
+      {
+        case ApiVersion::preview_v1:
+        default:
+        {
+          auto constitution_handle =
+            ctx.tx.template ro<ccf::Constitution>(ccf::Tables::CONSTITUTION);
+          auto constitution = constitution_handle->get();
+
+          if (!constitution.has_value())
+          {
+            detail::set_gov_error(
+              ctx.rpc_ctx,
+              HTTP_STATUS_NOT_FOUND,
+              ccf::errors::ResourceNotFound,
+              "Constitution not found");
+            return;
+          }
+
+          // Return raw JS constitution in body
+          ctx.rpc_ctx->set_response_status(HTTP_STATUS_OK);
+          ctx.rpc_ctx->set_response_body(std::move(constitution.value()));
+          ctx.rpc_ctx->set_response_header(
+            http::headers::CONTENT_TYPE,
+            http::headervalues::contenttype::JAVASCRIPT);
+          return;
+        }
+      }
+    };
+    registry
+      .make_read_only_endpoint(
+        "/service/constitution",
+        HTTP_GET,
+        api_version_adapter(get_constitution),
+        no_auth_required)
+      .set_openapi_hidden(true)
+      .install();
+
+    auto get_service_info = [&](auto& ctx, ApiVersion api_version) {
+      switch (api_version)
+      {
+        case ApiVersion::preview_v1:
+        default:
+        {
+          auto response_body = nlohmann::json::object();
+
+          auto service_info_handle =
+            ctx.tx.template ro<ccf::Service>(ccf::Tables::SERVICE);
+          auto service_info = service_info_handle->get();
+
+          if (!service_info.has_value())
+          {
+            detail::set_gov_error(
+              ctx.rpc_ctx,
+              HTTP_STATUS_NOT_FOUND,
+              ccf::errors::ResourceNotFound,
+              "Service info not yet available");
+            return;
+          }
+
+          response_body["status"] = service_info->status;
+          response_body["certificate"] = service_info->cert.str();
+          response_body["recoveryCount"] =
+            service_info->recovery_count.value_or(0);
+
+          if (service_info->current_service_create_txid.has_value())
+          {
+            response_body["creationTransactionId"] =
+              service_info->current_service_create_txid.value();
+          }
+          else
+          {
+            GOV_INFO_FMT("No recorded current_service_create_txid");
+          }
+
+          if (service_info->previous_service_identity_version.has_value())
+          {
+            // TODO: API definition is confused about the type of this. Decide
+            // whether this is a version, TxID, or cert
+            response_body["previousServiceCreationTransactionId"] =
+              service_info->previous_service_identity_version.value();
```

Comment:
Need to update TypeSpec type here to match implementation, and decide if we're returning a pointer to the previous ident, or the ident itself.